singleapplication_p.cpp

```diff
@@ -147,7 +147,14 @@ void SingleApplicationPrivate::genBlockServerName()
 #ifdef Q_OS_WIN
         appData.addData( SingleApplication::app_t::applicationFilePath().toLower().toUtf8() );
 #else
-        appData.addData( SingleApplication::app_t::applicationFilePath().toUtf8() );
+        QString appImagePath = qEnvironmentVariable( "APPIMAGE" );
```

Comment:
You don't need a QString, use `qgetenv`:

```suggestion
        const QByteArray appImagePath = qgetenv( "APPIMAGE" );
```

Reply:
@mklein-de I think you also need to remove `.toUtf8()` call on `appImagePath` in else block.